python/pyarrow/_parquet.pyx

```diff
@@ -1296,6 +1296,7 @@ cdef logical_type_name_from_enum(ParquetLogicalTypeId type_):
         ParquetLogicalType_ENUM: 'ENUM',
         ParquetLogicalType_DECIMAL: 'DECIMAL',
         ParquetLogicalType_DATE: 'DATE',
+        ParquetLogicalType_FLOAT16: 'FLOAT16',
```

Comment:
Very small nitpick: this listing is not actually alphabetical right now, so in that case I wouldn't put float in between date and time. Maybe move it below after INT or before NONE